harness/tests/common/api/test_authentication.py

```diff
@@ -125,3 +127,24 @@ def test_auth_json_v1_upgrade() -> None:
         # Make sure we got exactly the master urls we expected.
         exp = {"http://firstmaster:80", "https://secondmaster:443", "http://thirdmaster:8080"}
         assert set(obj["masters"]) == exp, list(obj["masters"])
+
+
+def test_salt_and_hash() -> None:
+    assert authentication.salt_and_hash("") == ""
+    assert (
+        authentication.salt_and_hash("A05850A9-bcfe-4ea6-8363-7350ee604be7")
+        == "c7e7c0ffdcdb574914b3c36dc6deb37a37e5c36f25dabef323cf470041c122"
+        + "470b3beab157cb8c1624cab6dd9d8e27dd242a128e5899a5c8b7fb7a1b7c521778"
+    )
+
+
+def test_check_password_complexity() -> None:
```

Comment:
Please break this up into separate tests. I'm not personally worried about copying and pasting boilerplate in tests, but if you want to avoid it there's always `pytest.parameterize`.